core_chat_data/src/commonTest/kotlin/net/thechance/mena/core_chat/data/contacts/ContactsRepositoryImplTest.kt

```
@@ -0,0 +1,408 @@
+
+import assertk.assertThat
+import assertk.assertions.isEmpty
+import assertk.assertions.isEqualTo
+import assertk.assertions.isFalse
+import assertk.assertions.isTrue
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.mock.respond
+import io.ktor.http.ContentType
+import io.ktor.http.HttpHeaders
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.headersOf
+import kotlinx.coroutines.test.runTest
+import kotlinx.io.IOException
+import kotlinx.serialization.json.Json
+import net.thechance.mena.core_chat.data.contacts.dto.ContactDto
+import net.thechance.mena.core_chat.data.contacts.fakes.FakeContactsProvider
+import net.thechance.mena.core_chat.data.contacts.fakes.FakeDataStore
+import net.thechance.mena.core_chat.data.shared.dto.PagedDataDto
+import net.thechance.mena.core_chat.domain.entity.Contact
+import net.thechance.mena.core_chat.domain.exception.ContactSyncFailedException
+import net.thechance.mena.core_chat.domain.exception.ContactsFetchFailedException
+import net.thechance.mena.core_chat.domain.exception.ContactsPermissionDeniedException
+import net.thechance.mena.core_chat.domain.exception.DataStoreException
+import net.thechance.mena.core_chat.domain.exception.UnAuthorizedException
+import net.thechance.mena.core_chat.domain.exception.UnknownException
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertFailsWith
+import com.bilalazzam.contacts_provider.Contact as DeviceContact
+
+
+class ContactsRepositoryImplTest {
+    private val mockContactsProvider = FakeContactsProvider()
+    private val mockDataStore = FakeDataStore()
+    private val json = Json { ignoreUnknownKeys = true }
+    private val jsonHeaders = headersOf(
+        HttpHeaders.ContentType,
+        ContentType.Application.Json.toString()
+    )
+
+    private lateinit var httpClient: HttpClient
+    private lateinit var repository: ContactsRepositoryImpl
+    private val sampleContact = Contact(
```

Since you are not depending on returning mocked data, but a fake data source, I think It would be better to extract these into fake data samples file and use the mappers to convert it to `ContactDto` that the api return, so the data won`t be hardcoded, also to make it easier to change the tests samples later without failing the tests